libraries/chain/controller.cpp

```diff
@@ -3101,6 +3101,12 @@ signed_block_ptr controller::fetch_block_by_id( block_id_type id )const {
    return signed_block_ptr();
 }
 
+std::optional<signed_block_header> controller::fetch_block_header_by_id( block_id_type id )const {
+   auto state = my->fork_db.get_block(id);
+   if( state && state->block ) return state->header;
+   return my->blog.read_block_header_by_num( block_header::num_from_id(id) );
```

Comment:
Need to verify that the block id matches the one requested. See `fetch_block_by_id` implementation above.